bin/gem2rpm

```diff
@@ -26,11 +26,33 @@ end
 
 rest = options[:args]
 
-template = begin
-  Gem2Rpm::Template.find options[:template_file], :gem_file => rest[0]
-rescue Gem2Rpm::Template::TemplateError => e
-  $stderr.puts e
-  exit(1)
+template = nil
+if template_file.nil?
+  f = open('/etc/os-release', 'r') if File.exists?('/etc/os-release')
+  if f
+    f.read.split('\n').each do |line|
+      line.match(%r{^ID=(.*)$}) { |m| template_file=m[1] }
+    end
+    f.close
+    f = nil
+  end
+  if template_file.match? '^"opensuse'
+    $stderr.puts 'Using template opensuse on openSUSE variant'
+    template_file = 'opensuse'
+  end
+end
+if template_file.nil?
+  template = Gem2Rpm::TEMPLATE
+else
+  begin
+    f = open(template_file, 'r') if File.exists?(template_file)
+    f = open(File.join(Gem2Rpm.template_dir, template_file + '.spec.erb'), 'r') unless f
+  rescue Errno::ENOENT
+    $stderr.puts "Could not open template #{template_file}. Aborting"
+    exit(1)
+  end
+  template = f.read
+  f.close
```

Comment:
I'm not a fan of this style. I'd say it's better to use:
```suggestion
  template_file = File.join(Gem2Rpm.template_dir, template_file + '.spec.erb') unless File.exist?(template_file)
  begin
    template = File.read(template_file)
  rescue Errno::ENOENT
    $stderr.puts "Could not open template #{template_file}. Aborting"
    exit(1)
  end
```

This uses fewer variables and fewer changes to leak file descriptors. It probably won't matter that much for a script, but I always try to use a safe style so I don't need to think about it.